nifi-nar-bundles/nifi-morphlines-bundle/nifi-morphlines-processors/src/main/java/org/apache/nifi/processors/morphlines/ExecuteMorphline.java

```diff
@@ -0,0 +1,253 @@
+        .defaultValue("_attachment_body")
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+        .name("success")
+        .description("Relationship for success.")
+        .build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+        .name("failure")
+        .description("Relationship for failure of morphlines.")
+        .build();
+
+    public static final Relationship REL_ORIGINAL = new Relationship.Builder()
+        .name("original")
+        .description("Relationship for original flowfiles.")
+        .build();
+
+    private static final List<PropertyDescriptor> PROPERTIES = ImmutableList.<PropertyDescriptor>builder()
+        .add(MORPHLINES_FILE)
+        .add(MORPHLINES_ID)
+        .add(MORPHLINES_OUTPUT_FIELD)
+        .build();
+
+    private static final Set<Relationship> RELATIONSHIPS = ImmutableSet.<Relationship>builder()
+        .add(REL_SUCCESS)
+        .add(REL_FAILURE)
+        .build();
+
+    public PropertyValue morphlinesFileProperty;
+    public PropertyValue morphlinesIdProperty;
+    public PropertyValue morphlinesOutputFieldProperty;
+    public Map<String, PropertyValue> dynamicPropertyMap = new HashMap();
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @Override
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        return new PropertyDescriptor.Builder()
+            .required(false)
+            .name(propertyDescriptorName)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .dynamic(true)
+            .expressionLanguageSupported(true)
+            .build();
+    }
+
+    @OnScheduled
+    public void onScheduled(ProcessContext context) throws Exception {
+        morphlinesFileProperty = context.getProperty(MORPHLINES_FILE);
+        morphlinesIdProperty = context.getProperty(MORPHLINES_ID);
+        morphlinesOutputFieldProperty = context.getProperty(MORPHLINES_OUTPUT_FIELD);
+        for (final PropertyDescriptor descriptor : context.getProperties().keySet()) {
+            if (descriptor.isDynamic()) {
+                dynamicPropertyMap.put(descriptor.getName(), context.getProperty(descriptor));
+            }
+        }
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if ( flowFile == null ) {
+            return;
+        }
+
+        FlowFile originalFlowFile = session.clone(flowFile);
+        final AtomicLong written = new AtomicLong(0L);
+        final byte[] value = new byte[(int) flowFile.getSize()];

```

Comment:
This is not going to work for big file

Reply:
It depends on your definition of "big" files. I believe that is not in the user interest to have big files (> 1 GB) being processed through the data flow as most processors under perform with this kind of files. So I do not think it is an issue here; I have seen the processor getting tremendous results on files up to 0.5 GB.